src/main/java/seedu/address/model/task/NameContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,31 @@
+package seedu.address.model.task;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Task}'s {@code Name} matches any of the keywords given.
+ */
+public class NameContainsKeywordsPredicate implements Predicate<Task> {
+    private final List<String> keywords;
+
+    public NameContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Task person) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getName().fullName, keyword));
+    }
```

Comment:
I think there is another small typo here 😅 It may be more consistent to rename `person` as `task`. Sorry for not spotting this previously!

Reply:
Ahh I see, must have missed that out haha. Thanks for pointing that out! :-)